### PR TITLE
WINDOWS: fix pip activate command

### PR DESCRIPTION
**Pull request by mattip**
## Why are these changes needed?
Continuation of #22449 

Fix pip activation so something like this will not crash
```
ray.init(runtime_env={"pip": ["toolz", "requests"]})
```

Also enable test that hit this code path.

## Checks

- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Comment by mattip** [comment]
`test_pip_task`, which hits the code path, [passes](https://buildkite.com/ray-project/ray-builders-pr/builds/28364#edd96521-1403-46bd-967b-7d2ff236d31e/2897-6853). It seems neither #23461 (which made the timeout longer in `test_metrics`) nor #23462 (which made the timeout longer on `test_ray_init`) helped: both the tests are failing in this CI run. Neither are related to the runtime environment changes here.

File: python/ray/_private/runtime_env/pip.py [hunk 1]
```diff
@@ -62,7 +62,10 @@ def get_virtualenv_python(cls, target_dir: str) -> str:
     def get_virtualenv_activate_command(cls, target_dir: str) -> str:
         virtualenv_path = cls.get_virtualenv_path(target_dir)
         # TODO(SongGuyang): Support Windows
```
Comment (architkulkarni) [inline python/ray/_private/runtime_env/pip.py:64]: Can we get rid of the TODO comment?

**Review by architkulkarni — APPROVED** [review]
Thanks, looks good!

File: python/ray/_private/runtime_env/pip.py (hunk 1 above)
Comment (mattip) [reply python/ray/_private/runtime_env/pip.py:64]: :+1:

**Comment by mattip** [comment]
`test_placement_group_5` is failing on linux like in other PR runs, everything else seems to pass? I don't know what is going on with the macos tests, they seem stuck

@pcmoritz could you take a look?

**Pull request merged**